crates/vault-registry/src/lib.rs

```diff
@@ -601,7 +601,12 @@ impl<T: Config> Module<T> {
             .to_vec(),
         )?;
         let mut vault = Self::get_rich_vault_from_id(&vault_id)?;
+        if vault.data.is_liquidated() {
+            Self::decrease_liquidation_vault_to_be_redeemed_tokens(tokens)?;
+        }
+
         vault.decrease_to_be_redeemed(tokens)?;
```

Comment:
I suggest to leave this as is, since #22 refactors this such that we _only_ do  `vault.decrease_to_be_redeemed(tokens)?;` (and then under the hood it's forwarded to the liquidation vault if required)